### PR TITLE
Add Metadata V2 Object and Interface 

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

In Preparation for Altair Hardfork

**What does this PR do? Why is it needed?**

- [x] Changes to a metadata interface to account for the multiple metadata types.
- [x] Adds in a new metadata v1 type in preparation for Altair.
- [x] Rename Metadata types for proper versioning.
- [x] Fix references across the repo.

**Which issues(s) does this PR fix?**

Part of #8638 

**Other notes for review**

### CODE REVIEW

File: proto/beacon/p2p/v1/messages.proto [hunk 1]
```diff
@@ -35,4 +35,18 @@ message ENRForkID {
...
+
+/*
+ Spec Definition:
+ MetaData
+ (
+ seq_number: uint64
+ attnets: Bitvector[ATTESTATION_SUBNET_COUNT]
+ )
+*/
+message MetaDataV2 {
```
Comment (terencechain) [inline proto/beacon/p2p/v1/messages.proto:48]: This is not related to this PR, at least by looking at the title and the description

File: shared/interfaces/metadata_wrapper.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
+package interfaces
+
+import (
+	"github.com/prysmaticlabs/go-bitfield"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// MetadataV1 is a convenience wrapper around our metadata protobuf object.
```
Comment (terencechain) [inline shared/interfaces/metadata_wrapper.go:9]:
```suggestion
// MetadataV0 is a convenience wrapper around our metadata protobuf object.
```

I wonder whether using `V0` and `V1` is better given we mostly start with versioning 0. `pb.MetaData` without any versioning makes me think it's closer to 0 than to 1 🤔

File: shared/interfaces/metadata_interface.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package interfaces
+
+import (
+	"github.com/prysmaticlabs/go-bitfield"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+// Metadata returns the interface of a p2p metadata type.
+type Metadata interface {
```
Comment (terencechain) [inline shared/interfaces/metadata_interface.go:9]: Do you want to add SyncBitfield() as well if you were to include V2 in the same PR

File: proto/beacon/p2p/v1/messages.proto (hunk 1 above)
Comment (nisdas) [reply proto/beacon/p2p/v1/messages.proto:48]: done

File: shared/interfaces/metadata_interface.go (hunk 3 above)
Comment (nisdas) [reply shared/interfaces/metadata_interface.go:9]: For the general metadata interface it should not have anything from V2.

File: shared/interfaces/metadata_wrapper.go (hunk 2 above)
Comment (nisdas) [reply shared/interfaces/metadata_wrapper.go:9]: good question lol, would technically have to change every reference in that case. I can try and changing it to a 0 and 1 to see how much work it would take.

File: shared/interfaces/metadata_interface.go (hunk 3 above)
Comment (nisdas) [reply shared/interfaces/metadata_interface.go:9]: I was thinking of adding this in our altair branch instead of here, given that is where it will be used.

**Review by terencechain — APPROVED** [review]
